volume/core.go

```diff
@@ -95,6 +98,29 @@ func StatParse(line string) (plugin.EntryAttributes, string, error) {
 // A DirMap is a map of directory names to a map of their children and the children's attributes.
 type DirMap = map[string]map[string]plugin.EntryAttributes
 
+func makedirs(attrs *map[string]map[string]plugin.EntryAttributes, newpath string) map[string]plugin.EntryAttributes {
```

Comment:
I think it'd be better if all the stat code was moved over to a separate `volume/stat.go` file because it clutters `core.go` and makes it harder to understand.

Reply:
I wasn't sure whether adding new entries to the map would be safe, and thought I was seeing an issue caused by it. I'll revert to using the reference; I think only arrays are different because you can create a new array with `append`.